include/SDL3_shadercross/SDL_shadercross.h

```diff
@@ -67,17 +67,21 @@ typedef struct SDL_ShaderCross_IOVarMetadata {
     Uint32 vector_size;                     /**< The number of components in the vector type of the variable. */
 } SDL_ShaderCross_IOVarMetadata;
 
-typedef struct SDL_ShaderCross_GraphicsShaderMetadata
+typedef struct SDL_ShaderCross_GraphicsShaderResourceInfo
 {
     Uint32 num_samplers;                     /**< The number of samplers defined in the shader. */
     Uint32 num_storage_textures;             /**< The number of storage textures defined in the shader. */
     Uint32 num_storage_buffers;              /**< The number of storage buffers defined in the shader. */
     Uint32 num_uniform_buffers;              /**< The number of uniform buffers defined in the shader. */
+} SDL_ShaderCross_GraphicsShaderResourceInfo;
+
+typedef struct SDL_ShaderCross_GraphicsShaderIOInfo
+{
     Uint32 num_inputs;                       /**< The number of inputs defined in the shader. */
     SDL_ShaderCross_IOVarMetadata *inputs;   /**< The inputs defined in the shader. */
     Uint32 num_outputs;                      /**< The number of outputs defined in the shader. */
     SDL_ShaderCross_IOVarMetadata *outputs;  /**< The outputs defined in the shader. */
-} SDL_ShaderCross_GraphicsShaderMetadata;
+} SDL_ShaderCross_GraphicsShaderIOInfo;
```

Comment:
I think `SDL_ShaderCross_GraphicsShaderMetadata` as an argument of the function that reflects graphics SPIRV is fine.

To avoid adding the extra reflect function, perhaps we can make `SDL_ShaderCross_GraphicsShaderResourceInfo` a member of `SDL_ShaderCross_GraphicsShaderMetadata`?

What do  you think? Or do you prefer the extra reflect function?

Reply:
> To avoid adding the extra reflect function, perhaps we can make SDL_ShaderCross_GraphicsShaderResourceInfo a member of SDL_ShaderCross_GraphicsShaderMetadata?

Hmm that's interesting. It might be nice to not have to set up the reflection structures twice... Maybe I'll try it in a separate branch.